cupyx/scipy/ndimage/filters.py

```diff
@@ -107,9 +107,17 @@ def _correlate_or_convolve(input, weights, output, mode, cval, origin,
         return output
     input = cupy.ascontiguousarray(input)
     weights = cupy.ascontiguousarray(weights, cupy.float64)
+
+    # The kernel needs only the non-zero kernel values and their coordinates.
+    # This allows us to use a single for loop to compute the ndim convolution.
+    # The loop will be over only the the non-zero entries of the filter.
+    wlocs = cupy.where(weights)
```

Comment:
This is equivalent to `cupy.nonzero(weights)` (I think it's better to write this way), so it synchronizes.

I wonder whether it's more significant in practice, compared to the improvement in this PR.

Reply:
`cupy.where` returns the mask of non-zero values, so device synchronization is not needed. Please note that `cupy.nonzero` synchronizes because it returns the indices of non-zero values.

Reply:
No, please check the documentation and the implementation:
https://github.com/cupy/cupy/blob/40b59e44d5bfe6296ce1b70ca54d121a0f304555/cupy/sorting/search.py#L191

Reply:
I see. it synchronizes as you say.